spec/test_spec.rb

```diff
@@ -0,0 +1,135 @@
+    end
+  end
+end
+
+RSpec.describe Book do
+  describe '#rent_book' do
+    let(:person) { Person.new('John Doe') }
+    let(:date) { Date.today }
+    let(:book) { Book.new('Title', 'Author') }
+
+    it 'creates a new rental and adds it to the rentals array' do
+      rental = book.rent_book(person, date)
+      expect(rental).to be_an_instance_of(Rental)
+      expect(book.rentals).to include(rental)
+      expect(person.rentals).to include(rental)
+    end
+  end
+
+  describe '#to_hash' do
+    let(:book) { Book.new('Title', 'Author') }
+
+    it 'returns a hash representation of the book' do
+      hash = book.to_hash
+      expect(hash).to be_a(Hash)
+      expect(hash).to eq({ title: 'Title', author: 'Author' })
+    end
+  end
+end
+
+RSpec.describe Decorator do
+  let(:nameable) { double('Nameable') }
+  subject { described_class.new(nameable) }
+
+  describe '#correct_name' do
+    it 'delegates the correct_name method to the nameable object' do
+      allow(nameable).to receive(:correct_name).and_return('John Doe')
+      expect(subject.correct_name).to eq('John Doe')
+    end
+  end
+end
+
+RSpec.describe CapitalizeDecorator do
+  let(:nameable) { double('Nameable') }
+  subject { described_class.new(nameable) }
+
+  describe '#correct_name' do
+    it 'capitalizes the name returned by the nameable object' do
+      allow(nameable).to receive(:correct_name).and_return('john doe')
+      expect(subject.correct_name).to eq('John doe')
+    end
+  end
+end
+
+RSpec.describe TrimmerDecorator do
+  let(:nameable) { double('Nameable') }
+  subject { described_class.new(nameable) }
+
+  describe '#correct_name' do
+    context 'when the name is longer than 10 characters' do
+      it 'trims the name to the first 10 characters' do
+        allow(nameable).to receive(:correct_name).and_return('Johnathan Doe')
+        expect(subject.correct_name).to eq('Johnathan ')
+      end
+    end
+
+    context 'when the name is not longer than 10 characters' do
+      it 'returns the name as is' do
+        allow(nameable).to receive(:correct_name).and_return('John Doe')
+        expect(subject.correct_name).to eq('John Doe')
+      end
+    end
+  end
+end
+
+RSpec.describe Classroom do
+  let(:classroom_label) { 'Mathematics' }
+  let(:classroom) { described_class.new(classroom_label) }
+  let(:student) { Student.new('John Doe') }
+
+  describe '#initialize' do
+    it 'sets the classroom label' do
+      expect(classroom.label).to eq(classroom_label)
+    end
+
+    it 'initializes an empty students array' do
+      expect(classroom.students).to be_empty
+    end
+  end
+
+  describe '#add_student' do
+    it 'adds a student to the students array' do
+      classroom.add_student(student)
+      expect(classroom.students).to include(student)
+    end
+
+    it 'sets the classroom for the added student' do
+      classroom.add_student(student)
+      expect(student.classroom).to eq(classroom)
+    end
+  end
+end
```

Comment:
- [ ] Kindly make sure there is one file per class, kindly put the Rental, Book, Classroom, Decorator... in thier own file for better structure.

your structure | a good structure
---|----
- [ ] Kindly make sure there is one file per class, kindly put the Rental, Book, Classroom, Decorator... in thier own file for better structure.

your structure | a good structure
---|----
<img width="205" alt="Screenshot 2023-07-06 at 00 10 45" src="https://github.com/federicaulzurrun/SchoolLibrary/assets/65068771/7567cb86-2289-4d63-a11a-18d51009d93d"> | <img width="223" alt="Screenshot 2023-07-06 at 00 09 21" src="https://github.com/federicaulzurrun/SchoolLibrary/assets/65068771/418140fa-e5bc-4883-89c0-049dcc7663db">